lib/ui/painting.dart

```diff
@@ -4314,13 +4314,27 @@ class FragmentShader extends Shader {
     _floats[index] = value;
   }
 
+  /// Sets the sampler uniform at [index] to [image].
+  ///
+  /// The index provided to setSampler is the index of the sampler uniform defined
+  /// in the fragment program, excluding all non-sampler uniforms.
+  ///
+  /// All the sampler uniforms that a shader expects must be provided or the
+  /// results will be undefined.
+  void setImageSampler(int index, Image image) {
+    assert(!debugDisposed, 'Tried to access uniforms on a disposed Shader: $this');
+    _setImageSampler(index, image);
+  }
+
   /// Sets the sampler uniform at [index] to [sampler].
   ///
   /// The index provided to setSampler is the index of the sampler uniform defined
   /// in the fragment program, excluding all non-sampler uniforms.
   ///
   /// All the sampler uniforms that a shader expects must be provided or the
   /// results will be undefined.
+  ///
+  /// Use of this method may result in poor performance with the Impeller backend.
```

Comment:
I would be careful about describing something as having poor performance with no refernce point. I've seen users go to extensive lengths to avoid invoking apis that have warnings like this, sometimes exceeding the original performance cost.

Reply:
SGTM, removed this.